src/modules/base/Module.sol

```diff
@@ -80,23 +80,30 @@ abstract contract Module is IModule, Initializable, ContextUpgradeable {
         bytes32 ownerRole = authorizer.getOwnerRole();
         bytes32 managerRole = authorizer.getManagerRole();
 
-        if (
-            !authorizer.hasRole(ownerRole, _msgSender())
-                && !authorizer.hasRole(managerRole, _msgSender())
-        ) {
-            revert Module__CallerNotAuthorized();
+        if (!authorizer.hasRole(managerRole, _msgSender())) {
+            revert Module__CallerNotAuthorized(managerRole, _msgSender());
+        } else if (!authorizer.hasRole(ownerRole, _msgSender())) {
+            revert Module__CallerNotAuthorized(ownerRole, _msgSender());
         }
         _;
     }
 
     /// @notice Modifier to guarantee function is only callable by addresses that hold a specific module-assigned role.
     modifier onlyModuleRole(uint8 roleId) {
         if (
-            !__Module_orchestrator.authorizer().isAuthorized(
-                roleId, _msgSender()
+            !__Module_orchestrator.authorizer().hasRole(
```

Comment:
Dont you skip with this the selfmanagedModules Step of the isAuthorized Function?

Reply:
Yes, but in this case you are specifically limiting the function to having that module role, so I think that it's warranted. Anyway, the auditors pointed out that the whole "toggleSelfManagement" only added unnecessary complexity at the end of the day, so I'm working on removing it

Reply:
In the meeting we had on Wednesday we agreed to make issues and PRs as granular as possible.
I think we should make this change in its own issue.

In general if you think the change of removing selfManagement is a useful one do it. I currently cant estimate how big the consequezes are. You probably have a better picture on it